satpy/readers/fci_l1c_fdhsi.py

```diff
@@ -131,42 +134,42 @@ def get_dataset(self, key, info=None):
 
         attrs = data.attrs.copy()
         info = info.copy()
+
         fv = attrs.pop(
-                "FillValue",
-                default_fillvals.get(data.dtype.str[1:], np.nan))
-        vr = attrs.pop("valid_range", [-np.inf, np.inf])
+            "FillValue",
+            default_fillvals.get(data.dtype.str[1:], np.nan))
+        vr = attrs.get("valid_range", [-np.inf, np.inf])
         if key.calibration == "counts":
             attrs["_FillValue"] = fv
             nfv = fv
         else:
             nfv = np.nan
         data = data.where(data >= vr[0], nfv)
         data = data.where(data <= vr[1], nfv)
-        if key.calibration == "counts":
-            # from package description, this just means not applying add_offset
-            # and scale_factor
-            attrs.pop("scale_factor")
-            attrs.pop("add_offset")
-            data.attrs["units"] = "1"
-            res = data
-        else:
-            data = (data * attrs.pop("scale_factor", 1) +
-                    attrs.pop("add_offset", 0))
-
-            if key.calibration in ("brightness_temperature", "reflectance"):
-                res = self.calibrate(data, key, measured, root)
-            else:
-                res = data
-                data.attrs["units"] = attrs["units"]
+
+        res = self.calibrate(data, key)
+
         # pre-calibration units no longer apply
         info.pop("units")
         attrs.pop("units")
 
         res.attrs.update(key.to_dict())
         res.attrs.update(info)
         res.attrs.update(attrs)
+
         res.attrs["platform_name"] = self._platform_name_translate.get(
                 self["/attr/platform"], self["/attr/platform"])
+
+        # remove unpacking parameters for calibrated data
+        if key.calibration in ['brightness_temperature', 'reflectance']:
+            res.attrs.pop("add_offset")
+            res.attrs.pop("warm_add_offset")
+            res.attrs.pop("scale_factor")
+            res.attrs.pop("warm_scale_factor")
+
+        # remove attributes from original file which don't apply anymore
+        res.attrs.pop('long_name')
+
         return res
 
     def get_channel_dataset(self, channel):
```

Comment:
Can this be renamed to `get_channel_dataset_key` for example ? I was confused and thought this was returning data. Maybe fix the corresponding docstring also.

Reply:
Good point, I simplified the function a bit and renamed it to get_channel_group_path